src/ucp/proto/proto_common.c

```diff
@@ -105,6 +101,12 @@ void ucp_proto_common_get_lane_distance(const ucp_proto_init_params_t *params,
     ucs_sys_device_t tl_sys_dev = context->tl_rscs[rsc_index].tl_rsc.sys_device;
     ucs_status_t status;
 
+    if (rsc_index == UCP_NULL_RESOURCE) {
+        distance->latency   = DBL_MAX;
+        distance->bandwidth = 0.;
```

Comment:
1. maybe set it to ucs_topo_default_distance?
IMO no real reason to report inf latency/0 bw here, can just reuse an existing default value
could also un-include float.h

2. need to remove line 101 to after checking (rsc_index == UCP_NULL_RESOURCE) to avoid out of bound access